src/utility/index.js

```diff
@@ -112,36 +112,78 @@ const getFirstKey = object => {
 /**
  * Return the name of the key (which is just the last word after the last period)
  *
- * @param string key    - key to use to return the name from
+ * @param {string} key    - key to use to return the name from
  *
- * @returns string name - the name of the key
+ * @returns {string} name - the name of the key
  */
-const getNameFromKey = key => key.split('.').pop();
+const getNameFromKey = key => key.split(KEY_SEPARATOR).pop();
 
 /**
  * Return the key of the parent (the key is everything before the last occurrence of a period)
  *
- * @param string key         - key to use to return the parent key from
+ * @param {string} key         - key to use to return the parent key from
  *
- * @returns string parentKey - the key of the parent
+ * @returns {string} parentKey - the key of the parent
  */
-const getParentKey = key => key.substr(0, key.lastIndexOf('.'));
+const getParentKey = key => key.substr(0, key.lastIndexOf(KEY_SEPARATOR));
+
+/**
+ * Gets the option keys from a complete key
+ *
+ * @param {string} key            - complete key to parse for option keys
+ *
+ * @returns {string[]} optionKeys - options keys parse from complete key
+ */
+const getOptionsKeysFromKey = key => {
+  const optionKeys = [];
+  let partialKey = key;
+  const re = new RegExp(SIMPLE_SCRIPT_OPTION_SEPARATOR, 'g');
+  let optionKey = partialKey
+    .substring(0, partialKey.lastIndexOf(SIMPLE_SCRIPT_OPTION_SEPARATOR))
+    .replace(re, KEY_SEPARATOR);
+  while (!isEmptyString(optionKey)) {
+    optionKeys.push(optionKey);
+    partialKey = partialKey.substring(0, partialKey.lastIndexOf(SIMPLE_SCRIPT_OPTION_SEPARATOR));
+    optionKey = partialKey
+      .substring(0, partialKey.lastIndexOf(SIMPLE_SCRIPT_OPTION_SEPARATOR))
+      .replace(re, KEY_SEPARATOR);
+  }
+  return optionKeys;
+};
+
+/**
+ * Returns true if script is a simple script and false otherwise
+ *
+ * @param {string} fileName          - name of file to check if is simple
+ *
+ * @returns {boolean} isSimpleScript - true if script is simple and false otherwise
+ */
+const isSimpleScript = fileName => fileName.split('.').includes(ScriptTypes.SIMPLE);
+
+/**
+ * Returns true if script is a advanced script and false otherwise
+ *
+ * @param {string} fileName            - name of file to check if is simple
+ *
+ * @returns {boolean} isAdvancedScript - true if script is advanced and false otherwise
+ */
+const isAdvancedScript = fileName => fileName.split('.').includes(ScriptTypes.ADVANCED);
```

Comment:
- [ ] Convert these two `isScript`-type functions into a single `getScriptType` function or just add the other function if these are really still needed anywhere